src/java/org/apache/cassandra/service/accord/async/AsyncWriter.java

```diff
@@ -238,7 +240,7 @@ private void denormalize(AccordCommand command, AsyncContext context, Object cal
         }
 
         // There won't be a txn to denormalize against until the command has been preaccepted
-        if (command.status().hasBeen(Status.PreAccepted) && AccordPartialCommand.WithDeps.serializer.needsUpdate(command))
+        if (command.status().hasBeen(Status.PreAccepted) && AccordPartialCommand.WithDeps.serializer.needsUpdate(command) && !(command.txn() == null && command.status().isInvalidated()))
```

Comment:
Not denormalizing an invalidated command may leave it in the uncommitted set, which could lead to incorrect deps being reported. That said, cfk denormalization may no longer be needed, and I'm going to try removing it in another branch, so hang tight

Reply:
if we do need to denormalize then we need these messages updated to pass along the `Keys`, as right now we don't have access

Reply:
yep makes sense, nevermind